File: backend/src/main/java/com/farid/backend/dto/ProductDTO.java
package com.farid.backend.dto;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.UUID;

@Data
@Builder
public class ProductDTO {
    private UUID id;
    private String name;
    private String description;
    private BigDecimal price;
    private String imageUrl;
}

File: frontend/src/app/model/ProductDTO.ts
import { UserDTO } from "./UserDTO";

export class ProductDTO{
    id: string;
    name: string;
    description: string;
    price: number;
    imageUrl: string;
    availableQuantity: number;
    categoryId: string;
    userDTO: UserDTO;

    constructor(
      id: string,
      name: string,
      description: string,
      price: number,
      imageUrl: string,
      availableQuantity: number,
      categoryId: string,
      userDTO: UserDTO
    ) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.price = price;
      this.imageUrl = imageUrl;
      this.availableQuantity = availableQuantity;
      this.categoryId = categoryId;
      this.userDTO = userDTO;
    }
  
}